sys/include/ng_net/ng_nettype.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net
+ * @{
+ *
+ * @file
+ * @brief   Protocol type definitions
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef NG_NETTYPE_H_
+#define NG_NETTYPE_H_
+
+/**
+ * @brief   Definition of protocol types in the network stack.
+ *
+ * @note    Expand at will.
+ */
+typedef struct {
+    NG_NETTYPE_UNDEF = 0,       /**< Protocol is undefined */
+    /**
+     * @{
+     * @name Link Layer
+     */
+#ifdef MODULE_NG_IEEE802154
+    NG_NETTYPE_IEEE802154,      /**< Protocol is IEEE 802.15.4 */
```

Comment:
this should also not be needed, as network interfaces are referenced by the interface id, right?